tests/UI/webservices/country/countryWs.ts

```diff
@@ -0,0 +1,76 @@
+import BaseWS from '@webservices/baseWs';
+import {APIRequestContext, APIResponse} from 'playwright';
+
+export default class CountryWS extends BaseWS {
+  public static endpoint = 'api/countries';
+
+  /**
+   * Get All Countries
+   * @param apiContext {APIRequestContext}
+   * @param authorization {string}
+   */
+  public static getAll(
+    apiContext: APIRequestContext,
+    authorization: string,
+  ): Promise<APIResponse> {
+    return super.getAll(apiContext, this.endpoint, authorization);
+  }
+
+  /**
+   * Get By id
+   * @param apiContext {APIRequestContext}
+   * @param authorization {string}
+   * @param idCountry {string}
+   */
+  public static getById(
+    apiContext: APIRequestContext,
+    authorization: string,
+    idCountry: string,
+  ): Promise<APIResponse> {
+    return super.getById(apiContext, this.endpoint, authorization, idCountry);
+  }
+
+  /**
+   * Add new Country
+   * @param apiContext {APIRequestContext}
+   * @param authorization {string}
+   * @param data {string} Xml of a new Country
+   */
+  public static add(
+    apiContext: APIRequestContext,
+    authorization: string,
+    data: string,
+  ): Promise<APIResponse> {
+    return super.add(apiContext, this.endpoint, authorization, data);
+  }
+
+  /**
+   * Update a country
+   * @param apiContext {APIRequestContext}
+   * @param authorization {string}
+   * @param idCountry {string}
+   * @param data {string} Xml of the new Country
+   */
+  public static update(
+    apiContext: APIRequestContext,
+    authorization: string,
+    idCountry: string,
+    data: string,
+  ): Promise<APIResponse> {
+    return super.update(apiContext, this.endpoint, authorization, idCountry, data);
+  }
+
+  /**
+   * Delete a country
+   * @param apiContext {APIRequestContext}
+   * @param authorization {string}
+   * @param idCountry {string}
+   */
+  public static delete(
+    apiContext: APIRequestContext,
+    authorization: string,
+    idCountry: string,
+  ): Promise<APIResponse> {
+    return super.delete(apiContext, this.endpoint, authorization, idCountry);
+  }
```

Comment:
You can remove it and move it to parent class.

Define endpoint in the parent class and override it in the current class.

Reply:
I'm not sure it's a good thing
if we removed it, you can only use one endpoint in a class
Example :  If we want to get the most used category on products, we will get all products, check most used category and get it by id. So it can't be one function on categoryWS, and if we start to import one WS class on another, we will have circular dependencies for sure.
So maybe it's a good thing to keep the option to pass a URL and not an endpoint